### PR TITLE
Update middleware callback to accept 4 arguments instead of 2 as in the base package.

### PR DESCRIPTION
**Pull request by tromgy**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

https://github.com/APIDevTools/swagger-express-middleware/blob/master/lib/middleware.js line 67

The JavaScript code this type definition is for has a callback that takes 4 arguments. The original type definition was typed as `(err: any, middleware: SwaggerMiddleware): void;`

Changed this to match JavaScript signature. Updated the test to demonstrate the use of the 3rd callback argument.

### CODE REVIEW

File: types/swagger-express-middleware/index.d.ts [hunk 1]
```diff
@@ -361,6 +363,11 @@ declare module "swagger-express-middleware" {
             externalDocs?: ExternalDocumentationObject
         }
 
+        export interface SwaggerParser {
```
Comment (uniqueiniquity) [inline types/swagger-express-middleware/index.d.ts]: Instead of defining your own interface for this, could you just use `@types/swagger-parser`?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/swagger-express-middleware/index.d.ts (hunk 1 above)
Comment (tromgy) [reply types/swagger-express-middleware/index.d.ts]: `SwaggerParser` is a namespace as defined in `@types/swagger-parser`, and so it doesn't work in the callback prototype `(err: any, middleware: SwaggerMiddleware, api: SwaggerObject, parser: SwaggerParser): void;` where a type is required.

Comment (uniqueiniquity) [reply types/swagger-express-middleware/index.d.ts]: ah, I see. In that case, you can use:
```ts
import * as parser from "swagger-parser";
type SwaggerParser = typeof parser.SwaggerParser;
```

Comment (tromgy) [reply types/swagger-express-middleware/index.d.ts]: Thanks! I was not aware of this trick of converting namespaces into types.

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**